src/framework-dfp.c

```diff
@@ -559,177 +1231,974 @@ int get_next_min_vertex(int fcg_stmt_offset, int stmt_id, int *list, int num, in
+ * not have a self edge in the FCG */
+int* get_common_parallel_dims(int scc_id, int* convex_successors, int num_convex_successors, int *colour, int current_colour, bool *is_colourable, PlutoProg *prog)
+{
+    int i, j, k, v;
+    Graph *ddg, *fcg;
+    Scc *sccs;
+    int scc_offset, succ_scc_offset, succ_scc;
+    int *common_dims;
+
+    ddg= prog->ddg;
+    fcg = prog->fcg;
+    sccs = ddg->sccs;
+
+    common_dims = NULL;
+    scc_offset = sccs[scc_id].fcg_scc_offset;
+    for (k=0; k<sccs[scc_id].max_dim; k++) {
+        if (colour[scc_offset+k]!=0 || !is_colourable[k]) continue;
+        for (i=0; i<num_convex_successors; i++) {
+            succ_scc = convex_successors[i];
+            succ_scc_offset = sccs[succ_scc].fcg_scc_offset;
+            for (j=0; j<sccs[succ_scc].max_dim; j++) {
+                v = succ_scc_offset+j;
+                /* The vertex can be coloured if there is no self edge on j, 
+                 * no edge between dimension j and k in the fcg, and there is 
+                 * no vertex adjecent to j that is already coloured. Also 
+                 * vertex j must be parallel and fusing with dimension k 
+                 * must not hinder parallelism */
+                if (colour[v]==0 && !fcg->adj->val[v][v] && ! is_adjecent(fcg, v, scc_offset+k) 
+                        && is_valid_colour (v, current_colour, fcg, colour) && 
+                        !par_preventing_adj_mat->val[v][v] && 
+                        !(par_preventing_adj_mat->val[v][scc_offset+k]|| par_preventing_adj_mat->val[scc_offset+k][v])) {
+                    if (common_dims==NULL) {
+                        common_dims = (int*) malloc (sizeof(int)*sccs[scc_id].max_dim);
+                        bzero(common_dims, sccs[scc_id].max_dim*sizeof(int));
+                    }
+                    /* If you have found one dimension in a convex successor then 
+                     * check the next convex successor. Hence the common_dims array 
+                     * indicates the number of successors that can be fused with the 
+                     * kth dimension of scc scc_id */
+                    common_dims[k] ++;
+                    break;
+                }
+            }
+        }
+    }
+    return common_dims;
+}
+
+/* Returns the dimension which has the maximum number of 
+ * convex successors. If there are none, then it returns -1  */
+int get_colouring_dim(int *common_dims, int max_dim)
+{
+    int i, dim, max;
+    max = 0;
+    dim = -1;
+    if (common_dims == NULL) {
+        return dim;
+    }
+    for (i=0;i<max_dim; i++) {
+        if (common_dims[i] > max) {
+            max = common_dims[i];
+            dim = i;
+        }
+    }
+    return dim;
+}
+
+
+/* colours sccs listed in convex_successors with the colour 
+ * current_colour. k is the vertex that was coloured previously */
+void colour_convex_successors(int k, int *convex_successors, int num_successors, int *colour, int current_colour, PlutoProg *prog)
```

Comment:
Wrap line to 80 columns - everywhere else as well as far as this pull request changes go.